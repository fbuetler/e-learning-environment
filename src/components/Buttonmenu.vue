<template>
  <div id="button-menu" class="flex-item flex-row flex-center flex-wrap">
    <button
      id="button-menu-next"
      class="card clickable"
      @click="emitRestartEvent()"
    >
      Nächstes Rätsel!
    </button>
    <button
      id="button-menu-check"
      class="card clickable"
      @click="emitEvalGameEvent()"
    >
      Überprüfen!
    </button>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Component } from "vue-property-decorator";
import { EventBus, EventBusEvents } from "@/components/EventBus";

@Component
export default class Buttonmenu extends Vue {
  emitRestartEvent() {
    EventBus.$emit(EventBusEvents.Restart);
  }
  emitEvalGameEvent() {
    EventBus.$emit(EventBusEvents.EvaluateGame);
  }
}
</script>

<style scoped></style>
