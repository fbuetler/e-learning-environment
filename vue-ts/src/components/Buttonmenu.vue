<template>
  <div class="buttons-container">
    <button @click="emitRestartGameEvent()">
      {{ restartGameText }}
    </button>
    <button @click="emitEvalGameEvent()">
      {{ evaluateGameText }}
    </button>
  </div>
</template>

<script lang="ts">
import Vue from "vue";
import { Component, Prop } from "vue-property-decorator";
import { EventBus, EventBusEvents } from "./EventBus";

@Component
export default class Buttonmenu extends Vue {
  @Prop({ type: String, required: true })
  private restartGameText!: string;
  @Prop({ type: String, required: true })
  private evaluateGameText!: string;

  private emitRestartGameEvent() {
    EventBus.$emit(EventBusEvents.RestartGame);
  }
  private emitEvalGameEvent() {
    EventBus.$emit(EventBusEvents.EvaluateGame);
  }
}
</script>

<style scoped>
input:focus,
select:focus,
textarea:focus,
button:focus {
  outline: none;
}

.buttons-container {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
}
button {
  border-radius: 6px;
  background-color: white;
  border: none;
  color: black;
  font-size: 16px;
  padding: 10px;
  width: 230px;
  cursor: pointer;
  font-weight: bold;
  margin: 0 1em 1em 1em;
}
</style>
