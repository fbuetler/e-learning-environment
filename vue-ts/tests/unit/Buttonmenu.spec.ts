import { shallowMount } from "@vue/test-utils";
import Buttonmenu from "@/components/Buttonmenu.vue";

describe("Buttonmenu.vue", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(Buttonmenu, {
      propsData: {
        restartGameText: "Restart",
        evaluateGameText: "Evaluate",
      },
    });
  });

  it("is a Vue instance)", () => {
    expect(wrapper.vm).toBeTruthy();
  });

  it("has buttons", () => {
    expect(wrapper.find("button").exists()).toBe(true);
  });

  it("correct button texts", () => {
    expect(wrapper.find("button#button-menu-next").text()).toBe("Restart");
    expect(wrapper.find("button#button-menu-check").text()).toBe("Evaluate");
  });

  it("button click restart", async () => {
    // TODO test eventBus events
    expect(true).toBeTruthy();
  });

  it("renders correctly", () => {
    expect(wrapper.element).toMatchSnapshot();
  });
});
