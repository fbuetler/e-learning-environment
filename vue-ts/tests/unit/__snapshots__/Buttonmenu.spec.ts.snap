// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttonmenu.vue renders correctly 1`] = `
<div
  class="flex-item flex-row flex-center flex-wrap"
>
  <button
    class="card clickable"
    id="button-menu-next"
  >
    
    Nächstes Rätsel!
  
  </button>
   
  <button
    class="card clickable"
    id="button-menu-check"
  >
    
    Überprüfen!
  
  </button>
</div>
`;
